src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationService.vb

```diff
@@ -135,9 +136,26 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
 
             ' Hide backing fields and events
 
+            ' GitHub #4208: When the user is typing a predicate (eg. "Enumerable.Range(0,10).Select($$")
+            ' we should filter out all the Func(Of delegate types, because they interfer with typing "Function".
+            Dim typeInferrer = context.GetLanguageService(Of ITypeInferenceService)()
+            Dim inferredTypes = typeInferrer.InferTypes(context.SemanticModel, context.Position, cancellationToken:=cancellationToken)
+            If inferredTypes?.Any(Function(t) t IsNot Nothing AndAlso IsFunc(t.OriginalDefinition)) Then
+                symbols = symbols.Where(Function(s) Not IsFunc(TryCast(s, ITypeSymbol)))
+            End If
+
             Return symbols.Where(Function(s) FilterEventsAndGeneratedSymbols(Nothing, s))
         End Function
 
+        Private Function IsFunc(t As ITypeSymbol) As Boolean
```

Comment:
Consider: somebody is using a library that also has delegate types that starts with "Func".